internal/searcher/client.go

```diff
@@ -55,13 +55,14 @@ func (c *client) SearchVersion(query, version string) (*SearchResult, error) {
 	)
 }
 
-func (c *client) Resolve(pkg, version string) (*lock.Package, error) {
-	result, err := c.SearchVersion(pkg, version)
+func (c *client) Resolve(pkg string) (*lock.Package, error) {
+	name, version, _ := strings.Cut(pkg, "@")
```

Comment:
Should we resolve versioned package only? What if name or version is missing?

Reply:
good call